### ISSUE TITLE
Wallet Connection Error (prompts phantom wallet lock screen twice)

### ISSUE THREAD

**Issue by haodev007** [solana-labs/wallet-adapter#519]
When I connect phantom wallet, it requires to open wallet(auto connect is true)
![image](https://user-images.githubusercontent.com/98631674/184965161-2d583aec-edda-4fa0-aa0c-fed18a3ce233.png)
Once I close it, it requries once again with error in the browser console.(but it is reject error)

![image](https://user-images.githubusercontent.com/98631674/184965942-13089b8f-d47b-4902-bf12-db699eb191ff.png)
When I close it again, it will be closed finally.

[Raydium Swap](https://raydium.io/) has also the same issue(I have checked it out).
Hope it will be fixed ASAP. and Appreciate any help and guide!

**Comment by jordansexton** [comment]
Without digging quite a bit, it'll be hard to tell how this is happening, and it might be application-specific and not something in the library. Can you provide some simple, reproducible code that does this, or can you look into and highlight what's causing this?

**Comment by haodev007** [comment]
Customized default wallet modal.
Anyways, using useWallet(select function, here) hook of @solana/wallet-adapter-react,

```typescript
...
const handleWalletClick = useCallback(
    (walletName: WalletName) => {
      select(walletName);
      setVisible(false); // hide wallet list modal
    },
    [select, setVisible]
  );
...

...
{installedWallets.map((wallet) => (
  <WalletListItem
    key={wallet.adapter.name}
    wallet={wallet}
    handleClick={() => handleWalletClick(wallet.adapter.name)}
  />
))}
{otherWallets.map((wallet) => (
  <WalletListItem
    key={wallet.adapter.name}
    wallet={wallet}
    handleClick={() => handleWalletClick(wallet.adapter.name)}
  />
))}
...

//WalletListItem.tsx
import React from "react";

import { WalletReadyState } from "@solana/wallet-adapter-base";

import { Image } from "components";
import { WalletListItemProps } from "types/wallet";

const WalletListItem = ({
  handleClick,
  tabIndex,
  wallet,
}: WalletListItemProps) => (
  <button
    tabIndex={tabIndex}
    onClick={handleClick}
    className="-m-3 flex items-center space-x-4 rounded-lg p-3 hover:bg-white hover:bg-opacity-5"
  >
    <Image
      src={wallet.adapter.icon}
      alt={wallet.adapter.name}
      width={24}
      height={24}
    />
    <div className="w-full">
      <div className="flex items-center justify-between text-base font-medium">
        {wallet.adapter.name}
        <span className="ml-auto mr-1 text-sm text-gray-400">
          {wallet.readyState === WalletReadyState.Installed && (
            <span>Detected</span>
          )}
        </span>
      </div>
    </div>
  </button>
);

export default WalletListItem;
```

Do you need more?

**Comment by Somnus007** [comment]
I have also encountered similar problems. When I click the `Select Wallet` button and select phantom, the phantom plug-in pops up, and then I click `Cancel` on the phantom plug-in, the current phantom plug-in will be closed, but then the pop-up window of the phantom plug-in will automatically pop up again.

**Comment by jordansexton** [comment]
Thanks! I'll try to find out what's happening.

**Comment by jordansexton** [comment]
I'm afraid I can't replicate this.

Here's a branch with the `react-ui-starter` project, using Phantom: https://github.com/solana-labs/wallet-adapter/blob/1e63b35ce0899d6f1cd78b38161df06f34ac5664/packages/starter/react-ui-starter/src/App.tsx#L33

Using this branch, run the instructions to build from source: https://github.com/solana-labs/wallet-adapter#build-from-source

> When I click the Select Wallet button and select phantom, the phantom plug-in pops up, and then I click Cancel on the phantom plug-in, the current phantom plug-in will be closed, but then the pop-up window of the phantom plug-in will automatically pop up again.

When I perform these steps, regardless of how I close the Phantom popup (by clicking the Cancel button, or closing the window), and regardless of whether Phantom is password locked or not, it doesn't pop up automatically again. I tried with and without autoConnect enabled.

Please make sure you're running the latest Phantom adapter and all other packages. If you are not, it could be the case that this was a bug in older versions that's been fixed?

**Comment by haodev007** [comment]
Hi, thanks for your kind reply.
I am running with these versions.
"@solana/wallet-adapter-base": "^0.9.7",
"@solana/wallet-adapter-react": "^0.15.10",
"@solana/wallet-adapter-react-ui": "^0.9.13",
"@solana/wallet-adapter-wallets": "^0.18.1",

**Comment by jordansexton** [comment]
So those are the versions in your package.json, but that doesn't mean you have those versions. You may need to wipe out your node_modules directory and install fresh.

**Comment by haodev007** [comment]
> So those are the versions in your package.json, but that doesn't mean you have those versions. You may need to wipe out your node_modules directory and install fresh.

Actually, I tried to reinstall node_modules several time. but still doesnt fix it

**Comment by jordansexton** [comment]
Can you post your yarn.lock file?

**Comment by Somnus007** [comment]
Hi @jordansexton , thanks for your reply. I just tried with a very [simple demo](https://github.com/Somnus007/solana-wallet-adapter-demo), I reproduced this again, feel free to try with it. Here is the screenshot. BTW, if I use `autoConnect=false` , I would not get this bug.

https://user-images.githubusercontent.com/31230649/185548536-370c508e-9424-432b-b934-9cfa6cc5ab0f.mp4

.

**Comment by jordansexton** [comment]
Thanks @Somnus007, can you tell me if this occurs for you with the branch linked here? https://github.com/solana-labs/wallet-adapter/issues/519#issuecomment-1219839618

I now suspect it may be something related to Next.js, perhaps something with server side rendering since that's the obvious major difference between Parcel and Next.

If it does not occur with Parcel, but it does occur with Next, can you try the Next.js starter project here? https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter/nextjs-starter

This will tell us if something about the Next configuration, or a newer version of Next, introduced the issue, or if we already have the issue in the Next.js starter project and simply haven't noticed.

**Comment by Somnus007** [comment]
@jordansexton , Just tried with both starters, 
- For the [#519 (comment)](https://github.com/solana-labs/wallet-adapter/issues/519#issuecomment-1219839618), it did NOT occur
- For the [Next.js starter](https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter/nextjs-starter), it happened again.
So I think this bug may be related to nextjs framework, hope this can help you.

**Comment by jordansexton** [comment]
I just merged https://github.com/solana-labs/wallet-adapter/pull/525 to fix what I think is an unrelated issue with autoconnect, but want to make sure.

Can I ask you to test one more thing? There's a new version of the React package out -- `@solana/wallet-adapter-react@0.15.14`

Can you try updating to this and running it again? If that doesn't fix it, it's definitely a Next thing, and I'll debug tomorrow.

**Comment by Somnus007** [comment]
It seems that an new problem occurred on `@solana/wallet-adapter-react@0.15.14` , I can not detect wallets now.
<img width="568" alt="screenshot" src="https://user-images.githubusercontent.com/31230649/185568953-8b069f91-d54f-47a5-8d22-4958f7bda123.png">

**Comment by jordansexton** [comment]
Hmm, that's not good! Can you tell me if the same thing occurs with Parcel? I just tested it with a Parcel project before releasing, so I want to narrow it to Next if possible.

**Comment by Somnus007** [comment]
It works well with Parcel ...

**Comment by jordansexton** [comment]
Hah, alright. So it's definitely a Next issue, but I'll get to the bottom of it. Thanks so much for narrowing it down.

**Comment by jordansexton** [comment]
Fixed in `@solana/wallet-adapter-react@0.15.15` and tested with the Next.js starter project.

**Comment by jordansexton** [comment]
This was a pretty subtle bug. Hard to say why it only happened in Next apps, but it wasn't Next-specific. Thanks for following through with report details!

**Comment by haodev007** [comment]
Thanks for your hard working. let me bump it

**Comment by haodev007** [comment]
> Fixed in `@solana/wallet-adapter-react@0.15.15` and tested with the Next.js starter project.

yet it works good. thanks!
